crates/h264/README.md

```diff
@@ -7,7 +7,231 @@
 
 ---
 
-A crate for encoding and decoding H.264 video headers.
+A pure Rust implementation of the H.264 encoder and decoder.
+
+This crate is designed to provide a simple and safe interface to encode and decode H.264 headers.
+
+## Why do we need this?
+
+This crate aims to provides a simple and safe interface for h264.
+
+## How is this different from other h264 crates?
+
+There are currently a handful of other active crates:
+
+- Retina
+  - This crate is heavily focused towards being as lightweight as possible since it's meant for security cameras. The downside of this is that it lacks some SPS fields. It is also lacking some documentation for the specific SPS fields.
+- less-avc
+  - This crate is focused on being a simple pure Rust implementation of H.264 but with only a subset of its features. IT lacks 4:4:4 support and some documentation for the specific SPS fields.
+- openh264-rs
+  - This crate is a set of bindings around the C implementation of H.264 (openh264). As a result, the crate contains a lot of unsafe code since it's just a lightweight rust wrapper.
+
+## Notable features
+
+This crate is a completely safe implementation of H264 encoding and decoding, which means there is no unsafe code!
+
+When combined with [scuffle-expgolomb](https://crates.io/crates/scuffle-expgolomb) and [scuffle-bytes-util](https://crates.io/crates/scuffle-bytes-util), working with h264 has never been easier!
```

Comment:
The goal of this library is to be able to parse / work with the config to deal in the various containers. Maybe at a later point we can include a full encoder / decoder for h264 but for the moment it is only headers.